### ISSUE TITLE
feat: Better Helm design

### ISSUE THREAD

**Issue by s0uky** [envoyproxy/gateway#4159]
*Description*:
The official Helm chart for the Gateway component should be more robust and comprehensive. Currently, it seems to lack resources for managing deployments of GatewayClass, Gateway, and other child objects associated with CRDs. Values file should be more configurable.

Extract all Kubernetes resource definitions related to the Gateway component from the source code (like this example with statsd: https://github.com/envoyproxy/gateway/blob/301eeddfa4c2f7e4e612874a9c5b13a976156c3e/internal/infrastructure/kubernetes/ratelimit/resource.go#L205 ) and consolidate them into a single location. This improves maintainability over Helm chart.

Ideally, the Helm chart should provide a single point of control to manage the creation and configuration of GatewayClass, Gateway, and other related CRD objects.

**Comment by zirain** [comment]
prom-statd-exporter issue tracked [here](https://github.com/envoyproxy/gateway/issues/3369).

**Comment by s0uky** [comment]
Yes, I know about the issue with statsd. My point was it's not good to hardcode any configuration which could be done through Helm.

**Comment by zirain** [comment]
indeed we should, that's a huge gap between rerfect and real.
any controbutions around this is welcome.
